### PR TITLE
Gh11244 Attribute Always Updatable

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/11244

### CODE REVIEW

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/handlingunits/HUEditorRowAttributes.java [hunk 1]
```diff
@@ -121,8 +121,9 @@ public static HUEditorRowAttributes cast(final IViewRowAttributes attributes)
 		for (final I_M_Attribute attribute : attributes)
 		{
 			final AttributeCode attributeCode = HUEditorRowAttributesHelper.extractAttributeCode(attribute);
+			final boolean alwaysUpdatable = attribute.isAlwaysUpdateable();
 
-			if (readonlyEffective || attributesStorage.isReadonlyUI(calloutCtx, attribute))
+			if ((!alwaysUpdatable && readonlyEffective) || attributesStorage.isReadonlyUI(calloutCtx, attribute))
```
Comment (metas-ts) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/handlingunits/HUEditorRowAttributes.java:126]: * i tried to draw a table, but this expression is too complicated for me please make it clearer
* readonlyEffective seems to be misnamed, same as `isReadonlyUI` (i mean, isReadonlyUI can return true and still it's writable, right?)

File: backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5591490_sys_gh11244_Attribute_IsAlwaysUpdatable.sql [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+;
+
+-- 2021-06-09T09:03:40.041Z
+-- URL zum Konzept
+INSERT INTO AD_Element (AD_Client_ID,AD_Element_ID,AD_Org_ID,Created,CreatedBy,Description,EntityType,IsActive,Name,PrintName,Updated,UpdatedBy) VALUES (0,579314,0,TO_TIMESTAMP('2021-06-09 12:03:39','YYYY-MM-DD HH24:MI:SS'),100,'When this flag is set on true it means the attribute''s value can be changed even after the handling unit is no longer in the "Palling" status.','D','Y','Attribute Always Updatable','Attribute Always Updatable',TO_TIMESTAMP('2021-06-09 12:03:39','YYYY-MM-DD HH24:MI:SS'),100)
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5591490_sys_gh11244_Attribute_IsAlwaysUpdatable.sql:53]: that's not true, as you documented. please find a description that tells the used what to expect and not expect when working with this flag

**Review by metas-ts — CHANGES_REQUESTED** [review]

**Comment by metas-ts** [comment]
PS: maybe M_Attribute.IsAlwaysUpdatable should be evaulated in `AbstractAttributeStorage#isReadonlyUI` to begin with

File: backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5591490_sys_gh11244_Attribute_IsAlwaysUpdatable.sql (hunk 2 above)
Comment (metas-rc) [reply backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5591490_sys_gh11244_Attribute_IsAlwaysUpdatable.sql:53]: done

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/handlingunits/HUEditorRowAttributes.java (hunk 1 above)
Comment (metas-rc) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/handlingunits/HUEditorRowAttributes.java]: rewritten, as talked

**Review by metas-ts — APPROVED** [review]

**Pull request merged**